sample/src/main/java/com/example/CashAppV3SampleActivity.kt

```diff
@@ -0,0 +1,284 @@
+                        showToast(this@CashAppV3SampleActivity, "Cash App Pay Exception")
+                        Log.e(tag, "Cash App Pay Exception", newState.exception)
+                    }
+
+                    Declined -> {
+                        showToast(this@CashAppV3SampleActivity, "Payment Declined")
+                    }
+
+                    NotStarted -> {
+                        bindings.cashappPayButton.isEnabled = false
+                    }
+
+                    CreatingCustomerRequest -> {
+                        // Use this state to display loading status if desired.
+                    }
+
+                    PollingTransactionStatus -> {
+                        // Use this state to display loading status if desired.
+                    }
+
+                    UpdatingCustomerRequest -> {
+                        // Use this state to display loading status if desired.
+                    }
+
+                    RetrievingExistingCustomerRequest -> {
+                        // Use this state to display loading status if desired.
+                        // Used only when retrieving existing customer request. Corner case (eg.: app killed)
+                    }
+
+                    Refreshing -> {
+                        // Optional. Use this state to display loading status if desired.
+                        bindings.cashappPayButton.isEnabled = false
+                    }
+                }
+            }
+        }
+
+    override fun onNewIntent(intent: Intent?) {
+        super.onNewIntent(intent)
+        /**
+         * If this activity was killed during Cash App flow, and recreated, then this intent
+         * can be used to start with an existing customer request. See
+         * Cash App Pay SDK for more information
+         * https://developers.cash.app/docs/api/technical-documentation/sdks/pay-kit/android-getting-started#start-with-an-existing-customer-request
+         *
+         * If this activity was not killed and instead just resumed then the [CashAppPayListener]
+         * above will continue to function and events should be handled there
+         */
+        Log.d(tag, "onNewIntent $intent  + ${intent?.extras}")
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        cashAppPay.registerForStateUpdates(cashAppPayListener)
+        bindings = CashAppV3LayoutBinding.inflate(LayoutInflater.from(this))
+        bindings.cashappPayButton.isEnabled = false
+
+        /**
+         * Step 0: Periodically check for new Merchant configurations. Doesn't need
+         * to happen before *every* transaction. Does need to happen before *first* transaction
+         * Use the tool of your choice to enable async non-main-thread network request (i.e. coroutines, rxjava)
+         *
+         * Using [lifecycleScope] is a quick-and-dirty example. Ideally you would not tie this network
+         * request to an Activity's lifecycle.
```

Comment:
Discussion point: although we purposely left any background work framework out of the SDK (coroutines, rxjava, etc), with the intent of making it play nicely with whatever each merchant is already using, WDYT about for the sample app making it opinionated (say VM + Coroutines). Reason why being, I've seen many merchants just copy-paste directly from the sample app - rather they copy something that is slightly more production ready 😅